Common/Cutscenes/DialogueState.cs

```diff
@@ -9,101 +9,189 @@
                 }
                 else // Hold the dialogue for the amount of time specified
                 {
-                    //Main.NewText("HOLD" + SecondaryTimer);
-                    if (SecondaryTimer++ <= player.DialogueList[0].showTime)
+                    if (DialogueTimer < player.DialogueList[0].drawTime) return;
+
+                    if (SecondaryTimer <= player.DialogueList[0].showTime)
                     {
-                        Dialogue.SetText(player.DialogueList[0].displayText);
+                        if (!Main.gamePaused) SecondaryTimer++;
 
-                        // Remove the dialogue from the list and reset counters
-                        if (SecondaryTimer == player.DialogueList[0].showTime)
+                        var text = player.DialogueList[0].displayText;
+
+                        if (player.DialogueList[0].bracketColor != null)
                         {
-                            Main.NewText("REMOVED");
+                            // Create a new string, adding in hex tags whenever an opening bracket is found
+                            var builder = new StringBuilder();
+                            builder.Append("    ");
+
+                            foreach (var character in text)
+                            {
+                                // Insert the hex tag if an opening bracket is found
+                                if (character == '[')
+                                {
+                                    builder.Append("[c/" + player.DialogueList[0].bracketColor + ":");
+                                }
+                                else
+                                {
+                                    builder.Append(character);
+                                }
+                            }
+
+                            if (!builder.ToString().Contains(']') && builder.ToString().Contains('[')) builder.Append(']');
+
+                            text = builder.ToString();
+                        }
 
+                        int hoveredSnippet = 0;
+                        TextSnippet[] snippets = ChatManager.ParseMessage(text, Color.White).ToArray();
+                        ChatManager.DrawColorCodedString(spriteBatch, FontAssets.MouseText.Value, snippets, textPosition, Color.White, 0f, Vector2.Zero, Vector2.One * 0.8f, out hoveredSnippet, MAXIMUM_LENGTH);
+                    }
+                    else
+                    {
+                        if (!Main.gamePaused) CloseTimer++;
+
+                        if (!player.DialogueList[0].closeAnimation) CloseTimer = (int)CLOSE_TIME;
+
+                        // Remove the dialogue from the list and reset counters
+                        if (CloseTimer == CLOSE_TIME)
+                        {
                             player.DialogueList.RemoveAt(0);
```

Comment:
So, the way you are essentially working with this is like a queue:

 - Push to the back of DialogueList
 - Read from it until exhausted.
 - Remove it

If you ever find yourself making a queue, you should often consider actually _using_ a `Queue`, which is a C# standard collection.

In this case, perhaps it would be neater to have a queue that you pop from the end of and assign to an "ActiveDialogueLine" field, or something along those lines? Might be clearer.

You could even stick it in a separate class, "ActiveDialogueState", containing `DialogueTimer`, `SecondaryTimer`, `OpenTimer` and `CloseTimer`, then create a new instance when you are picking a new line, that way you don't have to remember to reset it at the end of the method.